starter-code/src/clue.js

```diff
@@ -1,94 +1,152 @@
+  age: 22,
+  image: "https://metrouk2.files.wordpress.com/2016/07/professor-plum.jpg",
+  occupation: "Designer"
+};
+
+var missScarlet = {
+  first_name: "Kasandra",
+  last_name: "Scarlet",
+  color: "red",
+  description: "She is an A-list movie star with a dark past",
+  age: 31,
+  image: "https://metrouk2.files.wordpress.com/2016/07/miss-scarlett.jpg",
+  occupation: "Actor"
+};
 
-name: rope        ---   weight: 10
-name: knife       ---   weight: 8
-name: candlestick ---   weight: 2
-name: dumbbell    ---   weight: 30
-name: poison      ---   weight: 2
-name: axe         ---   weight: 15
-name: bat         ---   weight: 13
-name: trophy      ---   weight: 25
-name: pistol      ---   weight: 20
+var mrsPeacock = {
+  first_name: "Eleanor",
+  last_name: "Peacock",
+  color: "blue",
+  description:
+    "She is from a wealthy family and uses her status and money to earn popularity",
+  age: 36,
+  image: "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+  occupation: "Socialité"
+};
+
+var mrMustard = {
+  first_name: "Jack",
+  last_name: "Mustard",
+  color: "yellow",
+  description:
+    "He is a former football player who tries to get by on his former glory",
+  age: 62,
+  image: "https://metrouk2.files.wordpress.com/2016/07/colonel-mustard.jpg",
+  occupation: "Retired Football player"
+};
+
+// Weapons
+var weaponsArray = [
+  { name: "rope", weight: 10 },
+  { name: "knife", weight: 8 },
+  { name: "candlestick", weight: 2 },
+  { name: "dumbbell", weight: 30 },
+  { name: "poison", weight: 2 },
+  { name: "axe", weight: 15 },
+  { name: "bat", weight: 13 },
+  { name: "trophy", weight: 25 },
+  { name: "pistol", weight: 20 }
+];
 
 // Rooms
-name: Dinning Room
-name: Conservatory
-name: Kitchen
-name: Study
-name: Library
-name: Billiard Room
-name: Lounge
-name: Ballroom
-name: Hall
-name: Spa
-name: Living Room
-name: Observatory
-name: Theater
-name: Guest House
-name: Patio
-*/
+var roomsArray = [
+  "Dinning Room",
+  "Conservatory",
+  "Kitchen",
+  "Study",
+  "Library",
+  "Billiard Room",
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio"
+];
 
 // Characters Collection
 var charactersArray = [];
 
-// Rooms' Collection
-var roomsArray = [];
+charactersArray.push(
+  mrGreen,
+  drOrchid,
+  profPlum,
+  missScarlet,
+  mrsPeacock,
+  mrMustard
+);
+
+// console.log(charactersArray);
+
+function randomSelector(array) {
+  var random = Math.floor(Math.random() * array.length);
+  array.length;
+  return array[random];
+}
+var misteryEnvelope = [
+  [randomSelector(charactersArray).first_name],
+  [randomSelector(charactersArray).last_name],
+  [randomSelector(weaponsArray).name],
+  [randomSelector(roomsArray)]
+];
+
+function pickMistery() {
+  {
+    //  console.log([randomSelector(charactersArray).age])
+
+    return misteryEnvelope;
+  }
+}
+
+function revealMistery() {
+  console.log(
```

Comment:
you need to pass your misteryEnvelope to this function (and when you call it)

You then need to change some of the indexes- the first two should now have the same number (because they're from the same object) and then you need to target specially what you want 
```
misteryEnvelope[0].first_name
```